File: Source/Aladdin.ASN1.ISO/ASN1/ISO/OCSP/PreferredSignatureAlgorithms.cs
using System;
using System.Runtime.Serialization;

// PreferredSignatureAlgorithms ::= SEQUENCE OF PreferredSignatureAlgorithm

namespace Aladdin.ASN1.ISO.OCSP
{
	[Serializable]
	public class PreferredSignatureAlgorithms : Sequence<PreferredSignatureAlgorithm>
	{
		// конструктор при сериализации
        protected PreferredSignatureAlgorithms(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public PreferredSignatureAlgorithms(IEncodable encodable) : base(encodable) {}

		// конструктор при закодировании
		public PreferredSignatureAlgorithms(params PreferredSignatureAlgorithm[] values) : base(values) {}
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/ocsp/PreferredSignatureAlgorithms.java
package aladdin.asn1.iso.ocsp;
import aladdin.asn1.*;
import java.io.*;

// PreferredSignatureAlgorithms ::= SEQUENCE OF PreferredSignatureAlgorithm

public final class PreferredSignatureAlgorithms extends Sequence<PreferredSignatureAlgorithm>
{
    private static final long serialVersionUID = 7045176381862602231L;

    // конструктор при раскодировании
	public PreferredSignatureAlgorithms(IEncodable encodable) throws IOException 
	{
		super(encodable);
	}
	// конструктор при закодировании
	public PreferredSignatureAlgorithms(PreferredSignatureAlgorithm... values) 
	{
		super(PreferredSignatureAlgorithm.class, values); 
	}
}
